### ISSUE TITLE
Add package exclude to minimize JAR option.

### ISSUE THREAD

**Issue by Lauriethefish** [GradleUp/shadow#638]
While you can still use these methods, there might be cases where you would prefer excluding a package only rather than all of the dependency or a project's dependencies.
```gradle
// Force a class to be retained during minimization
shadowJar {
  minimize {
    exclude(dependency('org.scala-lang:.*:.*'))
  }
}
```

```gradle
shadowJar {
    minimize {
        // Includes transitive dependencies which may not always be ideal
        exclude(project(":bukkit"))
    }
}
```

### Shadow Version
6.1.0
### Gradle Version
6.8.2
### Expected Behavior
```gradle
shadowJar {
    minimize() {
        exclude 'com.lauriethefish.betterportals.*' 
    }
}    
```
Would include all classes within the package, and all the classes that those use, etc.

### Actual Behavior
Does nothing - package is removed.

### Build Script (final module that combines the others)
```gradle
plugins {
    id 'com.github.johnrengelman.shadow' version '6.1.0'
    id 'java'
}
sourceCompatibility = 1.8
targetCompatibility = 1.8

repositories {
    gradlePluginPortal()
    mavenCentral()
}

archivesBaseName = "BetterPortals-final"
version = '0.7.0'

apply plugin: 'com.github.johnrengelman.shadow'

dependencies {
    implementation project(':shared')
    implementation project(':bukkit')
    implementation project(':bungee')
}

tasks.build.dependsOn tasks.shadowJar
shadowJar {
    project.configurations.implementation.canBeResolved = true
    configurations = [project.configurations.implementation]

    relocate 'org.bstats', 'com.lauriethefish.bstats'

    minimize {
        exclude("com.lauriethefish.betterportals.*")
    }
}
```

### Content of Shadow JAR
https://gist.github.com/Lauriethefish/e0e125be42111d73cb9d1029b9c085e5

**Comment by sschuberth** [comment]
I have a similar use-case where we're using the [ServiceLoader](https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html) to load classes from our own code base / module. As these classes are not directly references, they get removed by `minimize()`. However, neither of the existing minimize-exclusion mechanisms fits: `dependency()` requires coordinates of a published artifact, and `project()` excludes whole modules.

What we'd need is a way to exclude single classes (ideally by specifying an interface excluded classes implement), or at last packages.

**Comment by sergeidyga** [comment]
Voting +1
In our case services are mentioned in `META-INF/services/<reference>`. So considering these references during `minimize()` might work as well.

(For now - didn't find better work-around then mentioning these files explicitly in the code)

**Comment by Goooler** [comment]
I don't think we should support excluding packages in `DependencyFilter`. `DependencyFilter` means it handles the work for dependencies. This feature request may rely on #1176 for a new API.
